### PR TITLE
Enable WebRTC

### PR DESCRIPTION
**Pull request by istankovic**
This PR adds the metadata querying during WebRTC signaling which was the last remaining thing needed to enable WebRTC.

### CODE REVIEW

File: raiden/network/transport/matrix/rtc/web_rtc.py [hunk 1]
```diff
@@ -22,19 +22,33 @@
...
+    Set,
+    Union,
+)
 
 log = structlog.get_logger(__name__)
 
 
 class _CoroutineHandler:
     def __init__(self) -> None:
+        super().__init__()
```
Comment (fredo) [inline raiden/network/transport/matrix/rtc/web_rtc.py:50]: Why is this necessary?

File: raiden/network/transport/matrix/rtc/web_rtc.py [hunk 2]
```diff
@@ -620,15 +604,17 @@ def stop(self) -> None:
         self._reset_state()
         self._send_hangup_messages()
 
+    def _send_signaling_message(self, address: Address, message: str) -> None:
+        metadata = _query_metadata(self._pfs_config, address)
```
Comment (fredo) [inline raiden/network/transport/matrix/rtc/web_rtc.py]: Is the private `_query_metadata` really necessary?

**Review by fredo — APPROVED** [review]
LGTM. Left some comments below. 

Additionally, in web_rtc.py in line 428 this needs to be changed to `via matrix to_device message` as room usage is outdated

File: raiden/network/transport/matrix/rtc/web_rtc.py (hunk 1 above)
Comment (istankovic) [reply raiden/network/transport/matrix/rtc/web_rtc.py:50]: Because `WebRTCManager` is derived from both `_CoroutineHandler` and `Runnable` and we need to call all the constructors.

File: raiden/network/transport/matrix/rtc/web_rtc.py (hunk 2 above)
Comment (istankovic) [reply raiden/network/transport/matrix/rtc/web_rtc.py]: Perhaps not as it is, but the idea is to add metadata caching for the duration of the signaling process, and then it will be nicer to have this factored out.

File: raiden/network/transport/matrix/rtc/web_rtc.py [hunk 3]
```diff
@@ -620,15 +603,17 @@ def stop(self) -> None:
         self._reset_state()
         self._send_hangup_messages()
```
Comment (fredo) [inline raiden/network/transport/matrix/rtc/web_rtc.py]: the second line will be a noop because `reset_state()` will empty the dictionary

File: raiden/network/transport/matrix/transport.py [hunk 4]
```diff
@@ -579,8 +583,7 @@ def stop(self) -> None:
         self._stop_event.set()
         self._broadcast_event.set()
 
-        if self._raiden_service:
-            self._web_rtc_manager.stop()
+        self._web_rtc_manager.stop()
```
Comment (fredo) [inline raiden/network/transport/matrix/transport.py:594]: I think the condition was there for a reason (although it was a false condition). Now, with the change that the web_rtc_manager is only initialized in `start()`, the argument for the if condition is even stronger.

If `stop()` is called before `start()` then `self.web_rtc_manager` will be none. I wonder why the linter does not prevent that.

**Review by fredo — CHANGES_REQUESTED** [review]
LGTM in general, but please have a look again if removing the if condition was the right thing to do.

File: raiden/network/transport/matrix/transport.py (hunk 4 above)
Comment (istankovic) [reply raiden/network/transport/matrix/transport.py:594]: If `stop` is called before `start`, it will exit immediately because of the `_stop_event` check. So it will all be good. In fact, I'm going to explicitly reset `_web_rtc_manager` to `None` to make it even more clear that it's valid only in the time period `<start, stop>`.

File: raiden/network/transport/matrix/rtc/web_rtc.py (hunk 3 above)
Comment (istankovic) [reply raiden/network/transport/matrix/rtc/web_rtc.py]: Indeed, you're right. What was the purpose of that before (my changes didn't touch that part)? It seems like those hangup messages were never sent. Shall I just remove that method?

Comment (fredo) [reply raiden/network/transport/matrix/rtc/web_rtc.py]: sounds reasonable. I think they should be sent as this is called in the stop method to notify the partners that the node is currently shutting down the channels
(In the past we had a failure in state showing that the datachannel was open although the partner already closed it)

Comment (fredo) [reply raiden/network/transport/matrix/rtc/web_rtc.py]: Maybe calling the hangup method before closing the channels makes sense

Comment (istankovic) [reply raiden/network/transport/matrix/rtc/web_rtc.py]: > Maybe calling the hangup method before closing the channels makes sense

Yes, I'll do that and see if it goes well.

**Review by fredo — APPROVED** [review]
